src/strategies/spidStrategy.ts

```diff
@@ -15,7 +15,11 @@ import {
 } from "../utils/idpLoader";
 import { log } from "../utils/logger";
 
-const IDP_IDS: { [key: string]: string | undefined } = {
+interface IIdpIds {
+  [key: string]: string | undefined;
```

Comment:
why `undefined` ? (you can use just  `Record<string, string>` instead of this interface)